pulp_rpm/app/tasks/publishing.py

```diff
@@ -211,6 +211,12 @@ def publish(repository_version_pk, metadata_signing_service=None, checksum_types
     """
     repository_version = RepositoryVersion.objects.get(pk=repository_version_pk)
     repository = repository_version.repository.cast()
+    checksum_types = checksum_types or {}
+    checksum_types["original"] = repository.original_checksum_types
```

Comment:
i know that maybe i am looking to far away in the future, but this code will not work if the content of the created repo was uploaded. `original_checksum_types` are set during sync, we would need to add same code to the upload workflow then. Also, during the migration of the repo these values are obviously not set

I was wondering if taking this approach would not be easier and more explicit:
```
if not checksum_types:
    checksum_types["package"] = CHECKSUM_TYPES.SHA256
    checksum_types["metadata"] = CHECKSUM_TYPES.SHA256
```
Also you can get rid of of the complex structure you  have in the 'original'
During sync it is enough to inspect only `primary' and take the checksum_type value from it, thend uring publish you can simplify the code here 

```
def get_checksum_types(checksum_types):
    """
    Get checksum algorithm for publishing metadata.

    Args:
        checksum_types (dict): Checksum types for metadata and packages.

    """
    original = checksum_types.get("original", CHECKSUM_TYPES.SHA256)
    metadata = checksum_types.get("metadata")

    if metadata:
        checksum_type = metadata
    else:
        checksum_type = original

    return getattr(cr, checksum_type.upper(), cr.SHA256)
```

Reply:
@fabricio-aguiar nevermind, let's just get rid of `package_checksum_type" so please remove this line:
```
original_package_checksum_type = repository.original_checksum_types.get(
        "package_checksum_type", CHECKSUM_TYPES.SHA256)
```

But edit this one:
```
            publication.package_checksum_type = checksum_types.get(
                "package", CHECKSUM_TYPES.SHA256)
```